File: auth-service/src/main/java/org/example/authservice/mapper/UserMapper.java
package org.example.authservice.mapper;

import org.example.authservice.dto.request.CreateUserRequest;
import org.example.authservice.dto.request.PatchUserRequest;
import org.example.authservice.dto.response.UserResponse;
import org.example.authservice.model.entity.User;
import org.mapstruct.BeanMapping;
import org.mapstruct.Mapper;
import org.mapstruct.MappingConstants;
import org.mapstruct.MappingTarget;
import org.mapstruct.NullValuePropertyMappingStrategy;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
public interface UserMapper {

    User toEntity(CreateUserRequest dto);

    UserResponse toResponse(User entity);

    void updateEntityFromDto(CreateUserRequest dto, @MappingTarget User entity);

    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    void patchEntityFromDto(PatchUserRequest dto, @MappingTarget User entity);
}

File: core-service/src/main/kotlin/by/modsen/libraryapp/mapper/UserMapper.kt
package by.modsen.libraryapp.mapper

import by.modsen.libraryapp.dto.response.UserResponse
import by.modsen.libraryapp.entity.User
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
interface UserMapper {

    fun toResponse(user: User): UserResponse
}